### DISCUSSION TITLE
Migrate from papertrail?

### DISCUSSION THREAD

**Discussion by jemminger** [palkan/logidze#178] [Q&A]
I currently use papertrail, and I want to delete its massively verbose table, but I don't want to lose the history.  

Is there a way I can migrate my papertrail versions over to logidze?

**Comment by palkan** [comment]
We don't have an out-of-the-box support for that.

The only complex part here is to convert papertrail history to [Logdize format](https://github.com/palkan/logidze#log-format).

And then you can just do `record.update_column(:log_data, converted_log)`.

**Comment by seanlinsley** [comment]
BTW, PaperTrail now supports dropping the `object` column, which reduces the table size by about half. Here's a migration you can use: https://github.com/paper-trail-gem/paper_trail/issues/1099#issuecomment-418460766

I considered Logidze as an alternative but the fact that all history is lost when a record is deleted (#61) isn't acceptable for long-lived Rails projects. For a greenfield project it's probably simple enough to disallow deletions.

**Comment by bf4** [comment]
Also noting deletion issue https://github.com/palkan/logidze/issues/61#issuecomment-526142844

**Comment by ziaulrehman40** [comment]
@seanlinsley Its a known and mentioned tradeoff, this has nothing to do with having a guide in place for migration to logidze.

I think that still should be done
